src/Test/Hspec/Expectations/Contrib.hs

```diff
@@ -24,3 +30,36 @@ isRight :: Either a b -> Bool
 isRight (Left  _) = False
 isRight (Right _) = True
 #endif
+
+-- |
+-- If you have a test case that has multiple assertions, you can use the
+-- 'annotate' function to provide a string message that will be attached to
+-- the 'Expectation'.
+--
+-- @
+-- describe "annotate" $ do
+--   it "adds the message" $ do
+--     annotate "obvious falsehood" $ do
+--       True `shouldBe` False
+--
+-- ========>
+--
+-- 1) annotate, adds the message
+--       obvious falsehood
+--       expected: False
+--        but got: True
+-- @
+--
+-- @since 0.8.3
+annotate :: String -> Expectation -> Expectation
+annotate msg = handle $ \(HUnitFailure loc exn) ->
+  throwIO $ HUnitFailure loc $ case exn of
+    Reason str ->
+      Reason $ msg ++
+          if null str then str else "\n" ++ str
+    ExpectedButGot mmsg expected got ->
+      let
+        mmsg' =
```

Comment:
More on the bikeshedding side of things, I generally avoid `'` in identifiers as they don't play well with `*`-search in Vim (+ adjusting it with e.g. `:set iskeyword+=$` breaks `*`-search for `c2hs`).

Reply:
Ah, good catch! I'll use a more meaningful name.